drivers/virtualbox/virtualbox.go

```diff
@@ -288,26 +289,37 @@ func (d *Driver) Create() error {
 		return err
 	}
 
-	// let VBoxService do nice magic automounting (when it's used)
-	if err := vbm("guestproperty", "set", d.MachineName, "/VirtualBox/GuestAdd/SharedFolders/MountPrefix", "/"); err != nil {
-		return err
-	}
-	if err := vbm("guestproperty", "set", d.MachineName, "/VirtualBox/GuestAdd/SharedFolders/MountDir", "/"); err != nil {
-		return err
-	}
+	var shareMountDir, shareDir, username string
 
-	var shareName, shareDir string // TODO configurable at some point
 	switch runtime.GOOS {
 	case "windows":
-		shareName = "c/Users"
-		shareDir = "c:\\Users"
+		username = os.Getenv("USERNAME")
+		shareDir = "c:\\Users\\" + username
+		shareMountDir = "/c/Users"
 	case "darwin":
-		shareName = "Users"
-		shareDir = "/Users"
-		// TODO "linux"
+		username = os.Getenv("USER")
+		shareDir = path.Join("/Users", username)
+		shareMountDir = "/Users"
```

Comment:
The replace seemed a bit hacky but I guess no different than what I currently have.  I do that.

`User.Current` does not work on Darwin which may be fine since we should be able to just use the path.

I think using the `homedir` and then replacing for msys sounds the best to me.  @tianon Your thoughts?